Banyan/src/clusters.jl

```diff
@@ -165,20 +177,43 @@ function get_clusters(; kwargs...)
             c["s3_read_write_resource"],
         ) for (name, c) in response["clusters"]
     )
+
+    # Cache info
+    global clusters
+    for (name, c) in clusters_dict
+        clusters[name] = c
+    end
+
+    return clusters_dict
+end
+
+function get_cluster_s3_bucket_arn(cluster_name=get_cluster_name(); kwargs...)
+    configure(; kwargs...)
+    global clusters
+    # Check if cached, sine this property is immutable
+    if !haskey(clusters, cluster_name)
+        get_cluster(cluster_name)
+    end
+    return clusters[cluster_name].s3_bucket_arn
 end
 
 function get_cluster_s3_bucket_name(cluster_name=get_cluster_name(); kwargs...)
     configure(; kwargs...)
-    cluster = get_cluster(cluster_name)
-    return s3_bucket_arn_to_name(cluster.s3_bucket_arn)
+    return s3_bucket_arn_to_name(get_cluster_s3_bucket_arn(cluster_name))
 end
 
-get_cluster(name::String=get_cluster_name(), kwargs...) = get_clusters(; kwargs...)[name]
+get_cluster(name::String=get_cluster_name(), kwargs...) = get_clusters(name; kwargs...)[name]
 
 get_running_clusters(args...; kwargs...) = filter(entry -> entry[2].status == :running, get_clusters(args...; kwargs...))
 
 function get_cluster_status(name::String=get_cluster_name(), kwargs...)
```

Comment:
we need to make sure that we are not caching cluster status if that is something that can be invalidtaed without the client side knowing. Are you sure we can't just get the cluster status once when we start creating the job and then pass that around to other functions that need it instead of calling `get_cluster_status` multiple times?